crates/tabby/src/services/completion/completion_prompt.rs

```diff
@@ -99,12 +104,17 @@ fn build_prefix(language: &str, prefix: &str, snippets: &[Snippet]) -> String {
     format!("{}\n{}", comments, prefix)
 }
 
-async fn collect_snippets(code: &dyn CodeSearch, language: &str, text: &str) -> Vec<Snippet> {
+async fn collect_snippets(
+    code: &dyn CodeSearch,
+    git_url: &str,
+    language: &str,
+    text: &str,
+) -> Vec<Snippet> {
     let mut ret = Vec::new();
     let mut tokens = tokenize_text(text);
 
     let serp = match code
-        .search_in_language(language, &tokens, MAX_SNIPPETS_TO_FETCH, 0)
+        .search_in_language(git_url, language, &tokens, MAX_SNIPPETS_TO_FETCH, 0)
```

Comment:
implement some heurstic based normalization to git url, for example:
1. git url being managed in repositories contains `https://github.com/TabbyML/tabby`
2. user's local git clone is from `https://github.com/TabbyML/tabby.git`

I suggest implement some herustic match like tokenization into [github.com, TabbyML, tabby], [github.com, TabbyML, tabby.git], and requires certain matching.

Reply:
Note: Use non-exact string match rather than sanitizing input